Add or update app-readme.md (#827)
rancher/charts: packages/rancher-monitoring/overlay/app-README.md

# Rancher Monitoring and Alerting

> This chart is based off of the upstream [kube-prometheus-stack](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack) chart. The chart deploys Prometheus operator and CRDs along with the Grafana chart, Prometheus adapter chart and additional charts to gather metrics. It allows users to monitor their Kubernetes clusters and view metrics in Grafana dashboards as well as set up alerts.

Reviewer: add links to relevant charts and added the fact that it also deploys k8s manifests.
```suggestion
This chart is based off of the upstream [kube-prometheus-stack](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack) chart. The chart deploys [Prometheus Operator](https://github.com/prometheus-operator/prometheus-operator) and its CRDs along with [Grafana](https://github.com/grafana/helm-charts/tree/main/charts/grafana), [Prometheus Adapter](https://github.com/prometheus-community/helm-charts) and additional charts / Kubernetes manifests to gather metrics. It allows users to monitor their Kubernetes clusters, view metrics in Grafana dashboards, and set up alerts and notifications.
```

Reviewer: The additional charts are node-exporter, kube-state-metrics, rancher-pushprox, etc. but I added the bit about k8s manifests since these are curated by the [kube-prometheus](https://github.com/prometheus-operator/kube-prometheus/) project but aren't explicitly in any Helm chart except `kube-prometheus-stack`.

For more information on how to use the feature, refer to our [docs](https://rancher.com/docs/rancher/v2.x/en/monitoring-alerting/v2.5/).

The chart installs the following components:

> - [Prometheus Operator](https://github.com/coreos/prometheus-operator)  - The operator provides easy monitoring definitions for Kubernetes services and manages [Prometheus](https://prometheus.io/) and [AlertManager](https://prometheus.io/docs/alerting/latest/alertmanager/) instances and deploys default monitors and alerts into the cluster.

Reviewer: Alertmanager isn't capitalized

```suggestion
- [Prometheus Operator](https://github.com/coreos/prometheus-operator)  - The operator provides easy monitoring definitions for Kubernetes services and manages [Prometheus](https://prometheus.io/) and [Alertmanager](https://prometheus.io/docs/alerting/latest/alertmanager/) instances and deploys default monitors and alerts into the cluster.
```

> - [Prometheus Operator](https://github.com/coreos/prometheus-operator)  - The operator provides easy monitoring definitions for Kubernetes services and manages [Prometheus](https://prometheus.io/) and [AlertManager](https://prometheus.io/docs/alerting/latest/alertmanager/) instances and deploys default monitors and alerts into the cluster.

Reviewer: Maybe this should be:
```suggestion
- [Prometheus Operator](https://github.com/coreos/prometheus-operator)  - The operator provides easy monitoring definitions for Kubernetes services, manages [Prometheus](https://prometheus.io/) and [AlertManager](https://prometheus.io/docs/alerting/latest/alertmanager/) instances, and adds default scrape targets for some Kubernetes components.
- [kube-prometheus](https://github.com/prometheus-operator/kube-prometheus/) - a collection of community-curated Kubernetes manifests, Grafana Dashboards, and PrometheusRules that deploy a default end-to-end cluster monitoring configuration.
```

# Rancher Monitoring and Alerting

>  This chart is based off of the upstream [kube-prometheus-stack](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack) chart. The chart deploys [Prometheus Operator](https://github.com/prometheus-operator/prometheus-operator) and its CRDs along with [Grafana](https://github.com/grafana/helm-charts/tree/main/charts/grafana), [Prometheus Adapter](https://github.com/prometheus-community/helm-charts) and additional charts / Kubernetes manifests to gather metrics. It allows users to monitor their Kubernetes clusters, view metrics in Grafana dashboards, and set up alerts and notifications.

Reviewer: nit: fix prometheus adapter link, minor grammar fix
```suggestion
 This chart is based on the upstream [kube-prometheus-stack](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack) chart. The chart deploys [Prometheus Operator](https://github.com/prometheus-operator/prometheus-operator) and its CRDs along with [Grafana](https://github.com/grafana/helm-charts/tree/main/charts/grafana), [Prometheus Adapter](https://github.com/prometheus-community/helm-charts/tree/main/charts/prometheus-adapter) and additional charts / Kubernetes manifests to gather metrics. It allows users to monitor their Kubernetes clusters, view metrics in Grafana dashboards, and set up alerts and notifications.
```

For more information on how to use the feature, refer to our [docs](https://rancher.com/docs/rancher/v2.x/en/monitoring-alerting/v2.5/).

The chart installs the following components:

- [Prometheus Operator](https://github.com/coreos/prometheus-operator)  - The operator provides easy monitoring definitions for Kubernetes services, manages [Prometheus](https://prometheus.io/) and [AlertManager](https://prometheus.io/docs/alerting/latest/alertmanager/) instances, and adds default scrape targets for some Kubernetes components.

> - [kube-prometheus](https://github.com/prometheus-operator/kube-prometheus/) - a collection of community-curated Kubernetes manifests, Grafana Dashboards, and PrometheusRules that deploy a default end-to-end cluster monitoring configuration.

Reviewer: nit: small capitalization fix
```suggestion
- [kube-prometheus](https://github.com/prometheus-operator/kube-prometheus/) - A collection of community-curated Kubernetes manifests, Grafana Dashboards, and PrometheusRules that deploy a default end-to-end cluster monitoring configuration.
```

> - [Grafana](https://github.com/helm/charts/tree/master/stable/grafana) - Grafana allows the ability to view dashboards of the Kubernetes cluster. The chart installs several default dashboards based on metrics available in the cluster.

Reviewer: nit: add in the ability to create dashboards. Also Grafana does not come with default dashboards; it's already mentioned that this comes from `kube-prometheus` above.

```suggestion
- [Grafana](https://github.com/helm/charts/tree/master/stable/grafana) - Grafana allows a user to create / view dashboards based on the cluster metrics collected by Prometheus.
```

> - [node-exporter](https://github.com/helm/charts/tree/master/stable/prometheus-node-exporter)/[kube-state-metrics](https://github.com/helm/charts/tree/master/stable/kube-state-metrics)/[rancher-pushprox](https://github.com/rancher/charts/tree/dev-v2.5/packages/rancher-pushprox/charts) - These components set up monitoring of internal Kubernetes components across different Kubernetes cluster types.

Reviewer: nit: on the rendered README, I think we need spaces between the slashes.

Also, node-exporter and kube-state-metrics don't monitor internal Kubernetes components (e.g. node-exporter monitors nodes / pods, kube-state-metrics monitors Kubernetes API objects), so I changed it to "various" instead.

```suggestion
- [node-exporter](https://github.com/helm/charts/tree/master/stable/prometheus-node-exporter) / [kube-state-metrics](https://github.com/helm/charts/tree/master/stable/kube-state-metrics) / [rancher-pushprox](https://github.com/rancher/charts/tree/dev-v2.5/packages/rancher-pushprox/charts) - These charts monitor various Kubernetes components across different Kubernetes cluster types.
```

> - [Prometheus Adapter](https://github.com/helm/charts/tree/master/stable/prometheus-adapter) - The adapter exposes custom metrics, resource metrics, and external metrics on the default [Prometheus](https://prometheus.io/) instance.

Reviewer: nit: we don't expose metrics by default, but a user has the ability to configure that with this deployed. By default, we only expose [certain container-based custom metrics](https://github.com/rancher/charts/blob/1d17a5bb6767f6468815b005b688fb73f7baa228/charts/rancher-monitoring/charts/prometheus-adapter/templates/custom-metrics-configmap.yaml#L16-L81).
```suggestion
- [Prometheus Adapter](https://github.com/helm/charts/tree/master/stable/prometheus-adapter) - The adapter allows a user to expose custom metrics, resource metrics, and external metrics on the default [Prometheus](https://prometheus.io/) instance to the Kubernetes API Server.
```

> For more information, review  the Helm README of this chart.

Reviewer: nit: seems like an extra space
```suggestion
For more information, review the Helm README of this chart.
```
